Library/Homebrew/download_strategy.rb

```diff
@@ -555,24 +557,93 @@ def curl(*args, **options)
   end
 end
 
-# Strategy for downloading a file from an GitHub Packages URL.
+# Strategy for downloading a file from a Docker registry.
 #
 # @api public
-class CurlGitHubPackagesDownloadStrategy < CurlDownloadStrategy
-  attr_writer :resolved_basename
+class DockerRegistryDownloadStrategy < CurlDownloadStrategy
+  attr_writer :mimetype, :resolved_basename
+
+  # Will store tokens for re-use
+  @@bearer_tokens = {} # rubocop:disable Style/ClassVars
 
   def initialize(url, name, version, **meta)
+    @auth_parser = Mechanize::HTTP::WWWAuthenticateParser.new
     meta ||= {}
     meta[:headers] ||= []
-    meta[:headers] << ["Authorization: Bearer QQ=="]
-    super(url, name, version, meta)
+    # Default "Accept" header value
+    @mimetype = "application/vnd.oci.image.index.v1+json"
```

Comment:
Any reason to not set this in `fetch` instead where it's used? Would avoid the need for an instance variable there too.

Reply:
Sorry, I don't mean an argument, I mean literally just moving this line into `def fetch` where it's used.